sdk/containerregistry/Azure.Containers.ContainerRegistry/api/Azure.Containers.ContainerRegistry.netstandard2.0.cs

```diff
@@ -1,181 +1,20 @@
 namespace Azure.Containers.ContainerRegistry
 {
-    public partial class AccessToken
+    public partial class ContainerRegistryClient
     {
-        internal AccessToken() { }
-        public string AccessTokenValue { get { throw null; } }
+        protected ContainerRegistryClient() { }
+        public ContainerRegistryClient(System.Uri endpoint, string username, string password) { }
```

Comment:
I wonder if `password` shouldn't use a `SecureString`. I know these are considered less-than-secure, but PowerShell itself does a number of things for SecureStrings that the CLR doesn't that might make it worth it for the Az PowerShell module. That said, if .NET is trying to push people away from SecureStrings (and they aren't the most usable), this probably makes sense.

At the same time, should this actually be replaced by a credential-like class? I know it can't be an actual `TokenCredential`, but even for API key credentials, we made a credential class that feels familiar. Hence, maybe you have your own `ContainerRegistryUsernamePasswordCredential` (to avoid confusing with Azure.Identity).

Reply:
I see later you commented about this, but I leave this unresolved for now (feel free to resolve it) for consideration.

Reply:
This functionality will be deprecated, so this[ constructor is temporary](https://github.com/azure/azure-sdk-for-net/issues/19469).  We won't ship it for preview.

(Incidentally, I had [designed a type for this](https://apiview.dev/Assemblies/Review/ba57bf904e524ce3bf647dad124ddc7c#Azure.Containers.ContainerRegistry.ContainerRegistryUserCredential), but we won't use it.